### PR TITLE
#19 mapbox

### PR DESCRIPTION
**Pull request by prokopishin**
Added mapbox
https://docs.mapbox.com/ios/maps/overview/
https://docs.mapbox.com/android/maps/overview/

### CODE REVIEW

File: buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -16,7 +16,7 @@ repositories {
 }
 
 dependencies {
-    implementation("dev.icerock:mobile-multiplatform:0.4.0")
+    implementation("dev.icerock:mobile-multiplatform:0.5.2")
```
Comment (Alex009) [inline buildSrc/build.gradle.kts]: please update to 0.5.3 - here fixed cocoapods integration

File: maps-google/build.gradle.kts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2019 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
+ * Copyright 2020 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
```
Comment (Alex009) [inline maps-google/build.gradle.kts]: it's invalid change :)

File: maps-mapbox/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
```
Comment (Alex009) [inline maps-mapbox/build.gradle.kts]: here should be 2020

File: maps-mapbox/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+ */
+
+plugins {
+    id("com.android.library")
+    id("org.jetbrains.kotlin.multiplatform")
+    id("kotlin-kapt")
+    id("kotlin-android-extensions")
+    id("dev.icerock.mobile.multiplatform")
+    id("maven-publish")
+    id(Deps.Plugins.kotlinSerialization.id)
```
Comment (Alex009) [inline maps-mapbox/build.gradle.kts]: serialization not used

File: maps-mapbox/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+plugins {
+    id("com.android.library")
+    id("org.jetbrains.kotlin.multiplatform")
+    id("kotlin-kapt")
```
Comment (Alex009) [inline maps-mapbox/build.gradle.kts]: it's used? maybe disable?

File: maps-mapbox/src/androidMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxRoute.kt [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+ * Copyright 2020 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package dev.icerock.moko.maps.mapbox
+
+import dev.icerock.moko.maps.MapElement
+
+actual class MapboxRoute : MapElement {
+    override fun delete() {
+        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
```
Comment (Alex009) [inline maps-mapbox/src/androidMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxRoute.kt]: not done?

File: maps-mapbox/src/androidMain/kotlin/dev/icerock/moko/maps/mapbox/Utils.kt [hunk 7]
```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright 2020 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package dev.icerock.moko.maps.mapbox
+
+import dev.icerock.moko.geo.LatLng
+
+internal fun LatLng.toAndroidLatLng() = com.mapbox.mapboxsdk.geometry.LatLng(
```
Comment (Alex009) [inline maps-mapbox/src/androidMain/kotlin/dev/icerock/moko/maps/mapbox/Utils.kt]: but created not `AndroidLatLang` - it `MapBoxLatLang`

File: sample/android-app/build.gradle.kts [hunk 8]
```diff
@@ -50,8 +50,8 @@ dependencies {
 
     implementation(Deps.Libs.Android.appCompat.name)
     implementation(Deps.Libs.Android.playServicesLocation.name)
-    implementation(Deps.Libs.Android.playServicesMaps.name)
-    implementation(Deps.Libs.Android.googleMapsServices.name)
+    implementation(Deps.Libs.Android.mapbox.name)
+    implementation(Deps.Libs.Android.mapboxAnnotation.name)
```
Comment (Alex009) [inline sample/android-app/build.gradle.kts]: not replace exist sample please - just add new

File: sample/android-app/src/main/java/com/icerockdev/app/MainActivity.kt [hunk 9]
```diff
@@ -6,49 +6,86 @@ package com.icerockdev.app
...
+class MainActivity : MvvmActivity<ActivityMainBinding, MapboxViewModel>() {
     override val layoutId: Int = R.layout.activity_main
     override val viewModelVariableId: Int = BR.viewModel
-    override val viewModelClass: Class<TrackerViewModel> = TrackerViewModel::class.java
+    override val viewModelClass: Class<MapboxViewModel> = MapboxViewModel::class.java
 
     override fun viewModelFactory(): ViewModelProvider.Factory {
         return createViewModelFactory {
-            TrackerViewModel(
+            MapboxViewModel(
```
Comment (Alex009) [inline sample/android-app/src/main/java/com/icerockdev/app/MainActivity.kt]: also not replace exist sample - add new sample!

File: buildSrc/build.gradle.kts (hunk 1 above)
Comment (prokopishin) [reply buildSrc/build.gradle.kts]: increased

File: maps-google/build.gradle.kts (hunk 2 above)
Comment (prokopishin) [reply maps-google/build.gradle.kts]: changed

File: maps-mapbox/build.gradle.kts (hunk 3 above)
Comment (prokopishin) [reply maps-mapbox/build.gradle.kts]: changed

File: maps-mapbox/build.gradle.kts (hunk 4 above)
Comment (prokopishin) [reply maps-mapbox/build.gradle.kts]: removed

File: maps-mapbox/build.gradle.kts (hunk 5 above)
Comment (prokopishin) [reply maps-mapbox/build.gradle.kts]: no, removed

File: maps-mapbox/src/androidMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxRoute.kt (hunk 6 above)
Comment (prokopishin) [reply maps-mapbox/src/androidMain/kotlin/dev/icerock/moko/maps/mapbox/MapboxRoute.kt]: routing has not yet been done on both platforms

File: maps-mapbox/src/androidMain/kotlin/dev/icerock/moko/maps/mapbox/Utils.kt (hunk 7 above)
Comment (prokopishin) [reply maps-mapbox/src/androidMain/kotlin/dev/icerock/moko/maps/mapbox/Utils.kt]: fixed

File: sample/android-app/build.gradle.kts (hunk 8 above)
Comment (prokopishin) [reply sample/android-app/build.gradle.kts]: ok, fixed

File: sample/android-app/src/main/java/com/icerockdev/app/MainActivity.kt (hunk 9 above)
Comment (prokopishin) [reply sample/android-app/src/main/java/com/icerockdev/app/MainActivity.kt]: fixed

**Pull request merged**